### PR TITLE
6886 add analysis modal

### PR DESCRIPTION
**Pull request by viddo**
Waypoint for #6886, containing the base + buffer and trade area types. 
The remaining work have some internal details that needs to be resolved, so we'll tackle them separately.

Merging back to integration branch, until the dependencies are merged to their master branches respectively.

@xavijam please review, have pair-programmed with @javierarce on this one, but a pair of extra :eyes: doesn't hurt

### CODE REVIEW

File: lib/assets/test/spec/cartodb3/editor/layers/layer-content-view.spec.js [hunk 1]
```diff
@@ -38,7 +36,7 @@ describe('editor/layers/layer-content-view', function () {
 
   it('should render correctly', function () {
     expect(_.size(this.view._subviews)).toBe(2);
-    expect(this.view.$('.Editor-HeaderInfo-TitleText').text()).toBe('hello');
+    expect(this.view.$('.Editor-HeaderInfo-TitleText').text()).toBe('foo');
```
Comment (xavijam) [inline lib/assets/test/spec/cartodb3/editor/layers/layer-content-view.spec.js]: oh! :+1:

File: lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js [hunk 2]
```diff
@@ -128,6 +140,10 @@ module.exports = Backbone.Collection.extend({
         layer.remove();
         this._createLayer(m);
       } else {
+        // TODO tmp hack until cartodb.js is updated
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: :scream:

**Comment by xavijam** [comment]
Not much to add, looks great :+1:.
Only the CartoDB.js update thing.

File: lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js (hunk 2 above)
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: Create [this issue](https://github.com/CartoDB/cartodb.js/issues/1216) to fix this.

Comment (viddo) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: Created an issue, https://github.com/CartoDB/cartodb.js/issues/1218, and updated the [postponed work issue](https://github.com/CartoDB/cartodb/issues/6950) to track this.

**Pull request merged**